### PR TITLE
Over JDBC we map Sets to PGArrays. Made collect_set return an array type.

### PR DESCRIPTION
**Pull request by andreidan**
## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [ ] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: sql/src/test/java/io/crate/integrationtests/WindowFunctionsIntegrationTest.java [hunk 1]
```diff
@@ -195,4 +195,22 @@ public void testLimitAndOffsetIsAppliedCorrectlyWith2DifferentWindowDefinitions(
                "50| 3.0| 40.0\n")
         );
     }
+
+    @Test
+    public void testCollectSetUnboundedFollowingWindowFrame() {
```
Comment (mfussenegger) [inline sql/src/test/java/io/crate/integrationtests/WindowFunctionsIntegrationTest.java]: Isn't this fix completely unrelated to the windowing function / frame? Would probably be more isolated to use `collect_set` as a global aggregate function?

**Comment by andreidan** [comment]
What do you think about dropping the Set type support from the parser as well? https://github.com/crate/crate/blob/master/sql-parser/src/main/antlr/SqlBase.g4#L539

**Comment by mfussenegger** [comment]
> What do you think about dropping the Set type support from the grammar as well? 

Sounds good

**Comment by andreidan** [comment]
@mfussenegger dropped the Set type from the grammar as well

File: sql/src/main/java/io/crate/execution/engine/aggregation/impl/CollectSetAggregation.java [hunk 2]
```diff
@@ -40,24 +41,33 @@
...
-public class CollectSetAggregation extends AggregationFunction<Set<Object>, Set<Object>> {
+public class CollectSetAggregation extends AggregationFunction<Set<Object>, Object[]> {
 
     public static final String NAME = "collect_set";
     private final SizeEstimator<Object> innerTypeEstimator;
 
-    private FunctionInfo info;
+    private final FunctionInfo info;
+    /**
+     * The partial return type is a SetType so we can maintain the java.util.Set intermediate type used for
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/execution/engine/aggregation/impl/CollectSetAggregation.java]: Not sure if this is the best place for this comment? The contract that the partialType is used for the intermediate state should imho be specified in `AggregationFunction`.  And the assignment to the mentioned set type isn't even happening here.

I'm also not sure if `returnType` is accurate. The partial data types are also used as inputs to the reduce function. `returnType` to me would imply a more narrow scope.

Comment (andreidan) [reply sql/src/main/java/io/crate/execution/engine/aggregation/impl/CollectSetAggregation.java]: Looking at the `AggregationFunction` I think the types and general functions documentation make this clear already, I'll drop this comment altogether.

Comment (andreidan) [reply sql/src/main/java/io/crate/execution/engine/aggregation/impl/CollectSetAggregation.java]: Thanks for the review and the suggestions. I'm quite happy with this cleanup and scope reduction for the SetType

**Pull request merged**